moveit_ros/move_group/src/move_group.cpp

```diff
@@ -287,9 +288,25 @@ int main(int argc, char** argv)
   // Initialize MoveItCpp
   const auto moveit_cpp = std::make_shared<moveit_cpp::MoveItCpp>(nh, moveit_cpp_options);
   const auto planning_scene_monitor = moveit_cpp->getPlanningSceneMonitorNonConst();
-
-  if (planning_scene_monitor->getPlanningScene())
+  if (auto ps = planning_scene_monitor->getPlanningScene())
   {
+    if (nh->has_parameter("default_planning_scene"))
+    {
+      std::string path = nh->get_parameter("default_planning_scene").as_string();
+      std::fstream file_stream;
+      file_stream.open(path, std::fstream::in);
+      if (!file_stream.is_open() || !ps->loadGeometryFromStream(file_stream))
```

Comment:
I think this is too dangerous because you're not locking the planning scene when writing to it (see https://github.com/moveit/moveit2/blob/19d4a2818bca7024305cbf5be5c3bca4186351d5/moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h#L170).
You can get thread safe access like this:
```cpp
{
  planning_scene_monitor::LockedPlanningSceneRW ps = planning_display_->getPlanningSceneRW();
  if (ps)
     FANCY MODIFICATIONS
} // End scope to release lock
```